### ISSUE TITLE
Packing from network location

### ISSUE THREAD

**Issue by jackjonez666** [mbruel/ngPost#12]
When packing from a network location on Windows, RAR fails due to a command line error, example:

Start Post #1: xxx.nzb
Compressing files: C:\Program Files\WinRAR\Rar.exe a -ep1 -m0 -idp -hpXxx -v1024m F:/Temp/pack/xxx.rar -r //tower2/disk10/XXX/XXX.mkv
ERROR: Unknown option: /tower2/disk10/XXX/XXX.mkv
=> rar exit code: 7

**Comment by mbruel** [comment]
Hi,
Try mounting the shared location on a disk.
You can do that from windows explorer, otherwise in command line [cf here](https://www.howtogeek.com/118452/how-to-map-network-drives-from-the-command-prompt-in-windows/)
Then import the files from that drive and that should work.
Please confirm and close the issue if that's working.
Thanks

**Comment by jackjonez666** [comment]
Yes, it does work, but I have 20 drives per server, mounting the main pool will spin up all 20 drives, so not ideal.  However, it does fix the issue, so thanks very much, issue closed!

**Comment by mbruel** [comment]
I don't have Windows but someone told me that if you change 
`C:\Program Files\WinRAR\Rar.exe a -ep1 -m0 -idp -hpXxx -v1024m F:/Temp/pack/xxx.rar -r //tower2/disk10/XXX/XXX.mkv`

to be
`C:\Program Files\WinRAR\Rar.exe a -ep1 -m0 -idp -hpXxx -v1024m F:/Temp/pack/xxx.rar -r \\tower2/disk10/XXX/XXX.mkv`

so instead of //tower2 \\tower2 it would work.
Can you give it a try on a cmd and let me know if it's true.
If so, I'll make a 4.2 quickly with the change, forcing to replace the first two slash with backslash.

**Comment by mbruel** [comment]
Are you using the Auto Posting Tab with the scanning button or just a Quick Post and file selection popup?
Please give a try on the Rar cmd using backslashes and let me know if it would work, I'll then release a new version by the end of the week

**Comment by r0x0r316** [comment]
normally that seems like a escaping bug and your approach of \\ instead of // should work. on the windows box i have no network shares so cannot test that right out of the box but usually command parameters in windows are given by - AND / so that is definitely the issue there. rar tries to interpet the //networkshare as /parameter. as i dont use windows that much i dont really know how to escape that. but a google search should resolve that quickly. it also isnt a buf in the uploader aswell since you already give a config option to resolve this issue by using a custom rar line.

**Comment by Uukrull** [comment]
I just tried it and works without problems:

rar.exe a -ep1 -m0 test.rar -r \\192.111.2.76/mnt/DRIVE/Test

**Comment by r0x0r316** [comment]
solved then.

**Comment by mbruel** [comment]
@Uukrull 
Well thanks Uukrull.
My question is how is it working in ngPost?
3 things I need to know:
  - if you do a Quick Post and Select a FILE that is on a shared folder. What do you see in the File list area (//shared_folder/fileName.ext ?)
   - if you do a Quick Post and Select a Folder. Idem what do you see in the list area.
   - Finally in the Auto Posting Tab, if you select a Shared folder. What do you see in the "Auto Dir path" \\\\ or //? If you press the scan button, same question for the items in the list.
Cheers ;)

**Comment by Uukrull** [comment]
> * if you do a Quick Post and Select a FILE that is on a shared folder. What do you see in the File list area (//shared_folder/fileName.ext ?)
> * if you do a Quick Post and Select a Folder. Idem what do you see in the list area.
![image](https://user-images.githubusercontent.com/5891591/73370082-f0844180-42b3-11ea-915a-8f70a97a1a50.png)


> * Finally in the Auto Posting Tab, if you select a Shared folder. What do you see in the "Auto Dir path" \\ or //? If you press the scan button, same question for the items in the list.

![image](https://user-images.githubusercontent.com/5891591/73370137-0560d500-42b4-11ea-80d3-005229edb5dc.png)

It seems Qt converts all `\` to `/`

**Comment by mbruel** [comment]
ok, I've pushed a fixed and compiled a win64 bit version. [it's here](https://files.fm/u/reckukk9)
Could someone try please.
I've made this change:
```
#if defined( Q_OS_WIN )
        {

            QString path = fileInfo.absoluteFilePath();
            if (path.startsWith("//"))
                path.replace(QRegExp("^//"), "\\");
            args << path;
        }
#else
            args << fileInfo.absoluteFilePath();
#endif
```
And I'm not sure if I should need to double the \\\\
I've try to print it and it was fine but I would have expect 4 backslashes instead of 2...

**Comment by jackjonez666** [comment]
Hello there, 

The above compiled version would need an additional "\" inserted and it should be working fine :-).

See below output:-

> Compressing files: C:\Program Files\WinRAR\Rar.exe a -ep1 -m0 -idp -hpA*H*T* -v100m F:\Temp\pack/bdr-bh62014test/bdr-bh62014test.rar \tower2/disk15/Share/movies/big hero 6 (2014)/big hero 6 (2014).mkv
> 
> RAR 5.80 x64   Copyright (c) 1993-2019 Alexander Roshal   5 Dec 2019
> Registered to SeVeN
> 
> Cannot open \tower2\disk15\Share\movies\big hero 6 (2014)\big hero 6 (2014).mkv
> 
> The system cannot find the path specified.

**Comment by mbruel** [comment]
yes it has been corrected.
[here is the latest version with the fix.](https://files.fm/u/23rvq5bf)

**Comment by jackjonez666** [comment]
Confirmed fixed in version 4.2, thank you!

Cheers,

-jj-
